src/main/kotlin/com/bugsnag/android/gradle/BugsnagPlugin.kt

```diff
@@ -346,11 +347,9 @@ class BugsnagPlugin : Plugin<Project> {
                 // DexGuard 9 runs as part of the bundle task supplied by AGP,
                 // so need to alter task dependency so that BAGP always runs
                 // after DexGuard
-                if (project.hasDexguardPlugin()) {
+                if (project.hasDexguardPlugin() && project.isDexguardEnabledForVariant(variant)) {
                     val taskName = getDexguardAabTaskName(variant)
-                    project.tasks.named(taskName).configure { dexguardTask ->
```

Comment:
It'd make sense to keep this configure block as otherwise I think this change could trigger eager configuration of the task

Reply:
The issue is that makes the order of `apply plugin` significant - the Dexguard plugin must be applied before BAGP otherwise the Dexguard task doesn't exist when we look it up here - also resulting in a `Task not found` exception.

By specifying the task by name rather than by reference we only resolve it at the end of configuration.

Reply:
Resolved this much more correctly by postponing the Dexguard task lookup using `generateProguardTaskProvider.configure`.